ehrql/query_model/table_schema.py

```diff
@@ -22,15 +23,24 @@ def __post_init__(self):
         def description(self):
             return f"Possible values: {', '.join(f'`{v}`' for v in self.values)}"
 
+        def validate(self, value):
+            return self.values == value if value else False
```

Comment:
This is not quite right.  A categorical column is one where values in the column should come from a given tuple (ie `self.values`).  So we should be validating whether the given value is in `self.values`.